MapleServer2/PacketHandlers/Game/UserSyncHandler.cs

```diff
@@ -49,15 +49,24 @@ public static void UpdatePlayer(GameSession session, SyncState[] syncStates)
         IFieldActor<Player> fieldPlayer = player.FieldPlayer;
 
         CoordF coord = syncStates[0].Coord.ToFloat();
-        CoordF closestBlock = Block.ClosestBlock(coord);
-        closestBlock.Z -= Block.BLOCK_SIZE; // Get block under player
 
-        if (IsCoordSafe(session, syncStates[0].Coord, closestBlock))
+        CoordF coordUnderneath = coord;
+        coordUnderneath.Z -= 50;
+
+        CoordF blockUnderneath = Block.ClosestBlock(coordUnderneath);
+        if (IsCoordSafe(player, syncStates[0].Coord, blockUnderneath))
         {
             CoordF safeBlock = Block.ClosestBlock(coord);
-            safeBlock.Z += 1; // Without this player will spawn inside the block
+            // TODO: Knowing the state of the player using the animation is probably not the correct way to do this
+            // we will need to know the state of the player for other things like counting time spent on ropes/running/walking/swimming
+            if (syncStates[0].Animation2 is 7 or 132) // swimming
```

Comment:
would be cool to have enums for each state so 7 and 132 don't need to be hard-coded, but doesn't need to be done in this PR

Reply:
Or even methods that take in a player and the states to returns booleans like `isSwimming` or `isRunning`